jerry-core/ecma/operations/ecma-function-object.c

```diff
@@ -129,24 +129,29 @@ ecma_op_is_callable (ecma_value_t value) /**< ecma value */
 } /* ecma_op_is_callable */
 
 /**
- * Checks whether the given object implements [[Construct]].
+ * Implement IsConstructor abstract operation.
  *
- * @return true - if the given object is constructor;
- *         false - otherwise
+ *
+ * @return NULL - if object is a valid for constructor call
+ *         char* - if object is not a valid constructor, the pointer contains the error message.
  */
-inline bool JERRY_ATTR_ALWAYS_INLINE
-ecma_object_is_constructor (ecma_object_t *obj_p) /**< ecma object */
+char *
+ecma_object_check_constructor (ecma_object_t *obj_p) /**< ecma object */
 {
   JERRY_ASSERT (!ecma_is_lexical_environment (obj_p));
 
   ecma_object_type_t type = ecma_get_object_type (obj_p);
 
-  if (type < ECMA_OBJECT_TYPE_PROXY)
+  if (JERRY_UNLIKELY (type < ECMA_OBJECT_TYPE_PROXY))
   {
-    return false;
+#if ENABLED (JERRY_ERROR_MESSAGES)
+    return "Invalid type for constructor call.";
+#else /* !ENABLED (JERRY_ERROR_MESSAGES) */
+    return "";
```

Comment:
What about using ECMA_ERR_MSG ? In all places. Should simplify the code a lot.

Reply:
ECMA_ERR_MSG returns NULL which in our case represents that there was no error.

Reply:
If it would be once, that would be ok, but there are so many similar cases. We should do something with it.

Reply:
Updated, now the `ECMA_ERR_MSG` returns an empty string if the error messages are disabled, and the messages passed for the raise_error* methods are checking for NULL and "" strings. Is this version ok? Or should I create another macro beside the `ECMA_ERR_MSG` which can return empty strings instead?

Reply:
"" may increase binary size. What about returning with `(char*)0x1` on success? We could define a macro for that.